akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/PathDirectivesSpec.scala

```diff
@@ -464,6 +464,75 @@ class PathDirectivesSpec extends RoutingSpec with Inside {
     }
   }
 
+  "the `ignoreTrailingSlash` directive" should {
+    var passes = 0
+    val route = ignoreTrailingSlash {
+      passes = passes + 1
```

Comment:
This might work right now but isn't strictly safe because neither routes nor tests are guaranteed to be executed synchronously on one thread. Maybe make it a `def route(counter: AtomicInteger)` or something so that the counter isn't shared between the test cases?